DevelopmentProcess.md

```diff
@@ -13,7 +13,13 @@ If you're a Project Leader, consider following these practices.
 
 ## Release Process
 
-*TBD([#33])*
+- [Semantic Versioning 2.0.0][semver] should be used for versioning project artifacts.
```

Comment:
lol, yeah. I screwed up the possession here. :D But it would probably sound better if we just shuffle the words like "project artifact versioning". I'm gonna address that later today. Thanks!